src/test/java/com/orzechazo/eshop/controllers/ProductControllerE2ETest.java

```diff
@@ -0,0 +1,167 @@
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+class ProductControllerE2ETest {
+    @Autowired
+    private ProductRepository productRepository;
+    MockMvc mockMvc;
+    private int DB_DEFAULT_PRODUCTS_SIZE;
+    private final ObjectWriter writer = new ObjectMapper().writer().withDefaultPrettyPrinter();
+    private final static String DB_PRODUCT_NAME = BootstrapProduct.DB_PRODUCT1_NAME;
+    BootstrapProduct bootstrap;
+    @BeforeEach
+    void setUp() {
+        bootstrap = new BootstrapProduct(productRepository);
+        bootstrap.loadData();
+
+        ProductServiceImpl productService = new ProductServiceImpl(productRepository);
+        ProductController productController = new ProductController(productService);
+        mockMvc = MockMvcBuilders.standaloneSetup(productController).build();
+        DB_DEFAULT_PRODUCTS_SIZE = bootstrap.getProducts().size();
+    }
+
+    @Test
+    void getAllProducts() throws Exception {
+        List<String> DB_PRODUCT_NAMES_LIST = bootstrap.getProducts().stream().map(Product::getName).toList();
+        mockMvc.perform(get("/products"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$",hasSize(DB_DEFAULT_PRODUCTS_SIZE)))
+                .andExpect(jsonPath("$[*].name").value(containsInAnyOrder(DB_PRODUCT_NAMES_LIST.toArray())));
+    }
+
+    @Test
+    void getProductByName() throws Exception {
+        mockMvc.perform(get("/products/" + DB_PRODUCT_NAME))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name",equalTo(DB_PRODUCT_NAME)))
+                .andExpect(jsonPath("$.amount",equalTo(5)))
+                .andExpect(jsonPath("$.netPrice",is(new BigDecimal("1.5").doubleValue())))
+                .andExpect(jsonPath("$.grossPrice",is(new BigDecimal("3.1").doubleValue())))
+                .andExpect(jsonPath("$.description",equalTo("testDescription1")));
+    }
+
+    @Test
+    void getProductByNameNotFound() throws Exception {
+        mockMvc.perform(get("/products/productNotInDb"))
```

Comment:
It is always good idea to extract path variables to constants and give them good names.